sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -218,7 +224,7 @@ public TextAnalyticsClient(Uri endpoint, AzureKeyCredential credential, TextAnal
         /// <exception cref="RequestFailedException">Service returned a non-success
         /// status code.</exception>
         public virtual async Task<Response<DetectLanguageResultCollection>> DetectLanguageBatchAsync(IEnumerable<string> documents, string countryHint = default, TextAnalyticsRequestOptions options = default, CancellationToken cancellationToken = default) =>
-            await _serviceClient.DetectLanguageBatchAsync(documents, countryHint, options, cancellationToken).ConfigureAwait(false);
+            await _serviceClient.DetectLanguageBatchAsync(documents, countryHint, options?.CheckSupported(ServiceVersion), cancellationToken).ConfigureAwait(false);
```

Comment:
What if we did this in post-facto validation and only if a service returned a 400 error?  That way, we only take the perf hit for validation if we're already invoking the exception framework.

Reply:
I like the idea. The only downside that came to mind is the customer is still incurring network IO. I would hope a bad request (or any non-2xx) wouldn't cost them either, but something to consider. @tg-msft, thoughts?

I wouldn't do it in this PR, but if we did it, in the subsequent change I talked about that, if necessary, could happen post-GA and service as a pattern (or even sharable, once moved) for other SDKs - maybe even DPGs.

Reply:
> What if we did this in post-facto validation and only if the service returned a 400 error? That way, we only take the perf hit for validation if we're already invoking the exception framework (and therefore in a known-bad performance case).

I don't think we can depend on a 400 in some cases.  For example, someone is doing a PUT request and has a model populated with properties that don't exist in the requested service version.  The service will silently ignore these properties - but they influence behavior in newer service versions.  As a result, the data is considered valid by the service, but the behavior does not match user expectations.

Reply:
This is being discussed in more detail by the REST API Stewardship board, but is beyond the scope of this PR. The version check is minimal and no more impactful than argument checks.